extensions/Push/lib/Connector/Phabricator.pm

```diff
@@ -0,0 +1,78 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# # License, v. 2.0. If a copy of the MPL was not distributed with this
+# # file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# #
+# # This Source Code Form is "Incompatible With Secondary Licenses", as
+# # defined by the Mozilla Public License, v. 2.0.
+
+package Bugzilla::Extension::Push::Connector::Phabricator;
+
+use 5.10.1;
+use strict;
+use warnings;
+
+use base 'Bugzilla::Extension::Push::Connector::Base';
+
+use Bugzilla::Bug;
+use Bugzilla::Constants;
+use Bugzilla::Extension::Push::Constants;
+use Bugzilla::Extension::Push::Util;
+use Bugzilla::User;
+
+use constant PHAB_CONTENT_TYPE => 'text/x-phabricator-request';
+use constant SECURITY_GROUP_KEY => '?'; # TODO:  Need to figure out what key this is
+
+# TODO:  Should we check if PhabBugz is enabled and bail if not?
+
+sub options {
+    return ({
+                name => 'phabricator_url',
+                label => 'Phabricator URL',
+                type => 'string',
+                default => '',
+                required => 1,
+            });
+}
+
+sub should_send {
+    my ($self, $message) = @_;
+
+    my $logger = Bugzilla->push_ext->logger;
+    $logger->info('should_send');
+
+    # TODO:  Ensure the specific security group is what was changed
+    my $did_change_security = ($message->routing_key =~ /^(?:attachment|bug)\.modify:.*\bis_private\b/);
```

Comment:
Here we cannot look for is_private specifically as that just signifies if the bug is has transitioned from public to private or private to public. We could have a bug that has an additional security group added or one or more removed but not yet public.

So we need to key off 'bug_group' I think which is the internal field name. We can then load the bug object in send() later to get the current list of set bug groups to compare against the ones we are syncing.